objects/client.c

```diff
@@ -363,151 +377,219 @@
  */
 
 /**
+ * Emitted when the client is raised within its layer.
+ *
  * @signal raised
+ * @see below
+ * @see above
+ * @see ontop
+ * @see raise
+ * @see lower
+ * @see lowered
  */
 
-/**
+/** Emitted when the client is lowered within its layer.
+ *
  * @signal lowered
+ * @see below
+ * @see above
+ * @see ontop
+ * @see raise
+ * @see lower
+ * @see raised
  */
 
 /**
  * The focused `client` or nil (in case there is none).
  *
+ * It is not recommanded to set the focused client using
+ * this property. Please use `c:activate{}` instead of
+ * `client.focus = c`. Setting the focus directly bypasses
+ * all the filters and emits fewer signals, which tend to
+ * cause unwanted side effects and make it harder to alter
+ * the code behavior in the future. It usually takes *more*
+ * code to use this rather than `:activate{}` because all
+ * the boilerplate code (such as `c:raise()`) needs to be
+ * added everywhere.
+ *
+ * The main use case for this field is to check *when* there
+ * is a focused client.
```

Comment:
I added the example. I am not sure how to generally handle this. `focus` and `active` are non-symmetrical concepts. I am not the one who started calling stuff `active` in the API, so I just deal with it. The *active* client must be the focused client, even if it is not on top. *Activating* a client is more than just focusing it. It involves potentially raising it, unminimizing it and maybe changing the current tag(s). These days (ever since v4.0), changing the focus using `client.focus` causes undesirable side effects.

Over the last years, I kept preferring the "active" word over the "focus" word across the API. Another PR, the phase 5 of the API standardization serie, adds `awful.client.focused()` to mirror `awful.screen.focused()` and formally deprecates `client.focus`.

"focus" would have probably been a better word all along, but its very late to fix that. Qt also has this non-symmetrical behavior in the API, but calls it focus either way.